Tribler/community/tunnel/conversion.py

```diff
@@ -58,13 +85,13 @@ def _decode_cell(self, placeholder, offset, data):
 
     def _encode_create(self, message):
         payload = message.payload
-        packet = pack("!IHH20s", payload.circuit_id, len(payload.node_public_key),
-                      len(payload.key), payload.node_id) + payload.node_public_key + payload.key
+        packet = pack("!IHH20s?", payload.circuit_id, len(payload.node_public_key),
+                      len(payload.key), payload.node_id, payload.exit_candidates) + payload.node_public_key + payload.key
```

Comment:
I don't have the details of the protocol in my head, but I really like if peers would choose the exit-node instead of relying on the n-1 peer. Additionally, it would remove the need of this exit_candidates flag